tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeExecutableImpl.kt

```diff
@@ -0,0 +1,67 @@
+package org.jetbrains.kotlin.gradle.plugin.experimental.internal
+
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.ConfigurationContainer
+import org.gradle.api.artifacts.ModuleVersionIdentifier
+import org.gradle.api.attributes.AttributeContainer
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.internal.component.SoftwareComponentInternal
+import org.gradle.api.internal.component.UsageContext
+import org.gradle.api.internal.file.FileOperations
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.language.cpp.internal.DefaultUsageContext
+import org.gradle.nativeplatform.Linkage
+import org.jetbrains.kotlin.gradle.plugin.experimental.KotlinNativeExecutable
+import org.jetbrains.kotlin.gradle.plugin.experimental.sourcesets.KotlinNativeSourceSet
+import org.jetbrains.kotlin.konan.target.CompilerOutputKind
+import javax.inject.Inject
+
+// TODO: SoftwareComponentInternal will be replaced with ComponentWithVariants by Gradle
+open class KotlinNativeExecutableImpl @Inject constructor(
+        name: String,
+        baseName: Provider<String>,
+        componentImplementation: Configuration,
+        sources: KotlinNativeSourceSet,
+        identity: KotlinNativeVariantIdentity,
+        objects: ObjectFactory,
+        projectLayout: ProjectLayout,
+        configurations: ConfigurationContainer,
+        fileOperations: FileOperations
+) : KotlinNativeBinaryImpl(name,
+        baseName,
+        sources,
+        identity,
+        projectLayout,
+        CompilerOutputKind.PROGRAM,
+        objects,
+        componentImplementation,
+        configurations,
+        fileOperations),
+        KotlinNativeExecutable,
+    SoftwareComponentInternal
```

Comment:
Wow that looks super akward..
I totally oversee when the constructor was closed...
Maybe use named parameters here as well?
And indend the two `interfaces`  the same way